README.md

```diff
@@ -1,5 +1,101 @@
+documentation](https://aws.amazon.com/elasticloadbalancing/applicationloadbalancer/).
+For an example of using ALB with ECS look no further than the [hashicorp example](https://github.com/terraform-providers/terraform-provider-aws/blob/master/examples/ecs-alb).
+
+## Input Variables
+* `alb_is_internal` - Determines if the ALB is externally facing or internal. (Optional; default: false)
+* `alb_name` - Name of the ALB as it appears in the AWS console. (Optional; default: my-alb)
+* `alb_protocols` - A comma delimited list of protocols the ALB will accept for incoming connections. Only HTTP and HTTPS are supported. (Optional; default: HTTPS)
+* `alb_security_groups` - A comma delimited list of security groups to attach to the ALB. (Required)
+* `aws_region` - Region to deploy our resources. (Required)
+* `backend_port` - Port on which the backing instances serve traffic. (Optional; default: 80)
+* `backend_protocol` - Protocol the backing instances use. (Optional; default: HTTP)
+* `certificate_arn` - . (Required if using HTTPS in `alb_protocols`)
+* `cookie_duration` - If sticky sessions via cookies are desired, set this variable to a value from 2 - 604800 seconds. (Optional)
+* `health_check_path` - Path for the load balancer to health check instances. (Optional; default: /)
+* `log_bucket` - S3 bucket where access logs should land. If not set it will be not leave logs. (Optional; default: "")
+* `log_prefix` - S3 prefix within the `log_bucket` where logs should land. (Optional; default: "")
+* `security_policy` - The security policy if using HTTPS externally on the ALB. See [AWS docs](https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/elb-security-policy-table.html).
+* `subnets` - ALB will be created in the subnets in this list. (Required)
+* `vpc_id` - Resources will be created in the VPC with this `id`. (Required)
+* `tags` - A mapping of tags to assign to the resource.
+
+## Outputs
+* `alb_id` - `id` of the ALB created.
+* `alb_dns_name` - DNS CNAME of the ALB created.
+* `alb_zone_id` - Route53 `zone_id` of the newly minted ALB.
+* `target_group_arn` - `arn` of the target group. Useful for passing to your Auto Scaling group module.
+* `principal_account_id` - the id of the AWS root user within this region. See [docs here]('http://docs.aws.amazon.com/elasticloadbalancing/latest/classic/enable-access-logs.html#attach-bucket-policy').
+
+## Usage example:
+A full example leveraging other community modules is contained in the [examples/test_fixtures directory](examples/test_fixtures). Here's the gist of using it via the Terraform registry:
+```
+module "alb" {
+  source              = "terraform-aws-modules/alb/aws"
+  alb_security_groups = "${var.alb_security_groups}"
+  certificate_arn     = "${var.certificate_arn}"
+  log_bucket          = "${var.log_bucket}"
+  log_prefix          = "${var.log_prefix}"
+  subnets             = "${var.public_subnets}"
+  vpc_id              = "${var.vpc_id}"
```

Comment:
I usually put the highest-impact attributes at the top. In this case vpc_id and subnets at the most required.

Use complete values instead of references to `data sources`, `var`, `other resources`. This example should show exactly what is expected, but not how you will pass it into.

Reply:
done.